throttler/throttler.go

```diff
@@ -0,0 +1,78 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+//ThrottledAction is a common interface for actions that can be executed
+type ThrottledAction interface {
+	Execute()
+}
```

Comment:
A `Runner` can run/execute anything regarding of how long.

Saying that I agree that the `server` concept seems to imply is long-running. That's why I propose to change the `server` name to `runner`.

Also, If this is supposed to be a `Runner` decorator, I would place it inside the `runner` package.

Anyway, maybe that's too different to what this PR provides. Anyway, my two cents!